policy/resource.go

```diff
@@ -26,12 +26,59 @@ import (
 	"github.com/minio/pkg/v3/wildcard"
 )
 
-// ResourceARNPrefix - resource ARN prefix as per AWS S3 specification.
-const ResourceARNPrefix = "arn:aws:s3:::"
+const (
+	// ResourceARNPrefix - resource S3 ARN prefix as per AWS S3 specification.
+	ResourceARNPrefix = "arn:aws:s3:::"
+
+	// ResourceARNKMSPrefix is for KMS key resources. MinIO specific API.
+	ResourceARNKMSPrefix = "arn:minio:kms:::"
+)
+
+// ResourceARNType - ARN prefix type
+type ResourceARNType uint32
+
+const (
+	// ResourceARNS3 is the ARN prefix type for AWS S3 resources.
+	ResourceARNS3 ResourceARNType = iota + 1
+
+	// ResourceARNKMS is the ARN prefix type for MinIO KMS resources.
+	ResourceARNKMS
+
+	unknownARN // for detecting errors
```

Comment:
Move this to value 0 - then it also catches the zero value. You can use `endARN` here, if you need the size anywhere.

That way `Resource{} ==Resource{Type: unknownARN}`, which seems reasonable.

Reply:
Sounds good, but I don't think we need the size? If I add it, I'll need to use it somewhere or disable linting on that line, as the linter complains that it's unused.

Reply:
I've moved to zero value but have not added endARN, LMK if I should and how you would like it used.